tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -0,0 +1,1078 @@
+        return methodUsage.getDeclaration().getTypeParameters()
+                .stream()
+                .map(ResolvedTypeParameterDeclaration::getName)
+                .toList();
+    }
+
+    /**
+     * Gets all formal parameters in the method definition. This method gets
+     * the type of each parameter, followed by an artificial name. For
+     * example,
+     *     "MethodUsage[get(int i)]"    => "List.of("int arg0")"
+     */
+    private static List<String> getParameters(MethodUsage methodUsage) {
+        ResolvedMethodDeclaration methodDeclaration = methodUsage.getDeclaration();
+        // iterate through each parameter in the method declaration.
+        List<String> methodParameters = new ArrayList<>();
+        for (int i = 0; i < methodDeclaration.getNumberOfParams(); i++) {
+            methodParameters.add(getTypeWithoutPackages(methodDeclaration.getParam(i).getType()) + " arg" + i);
+        }
+        return methodParameters;
+    }
+
+    /**
+     * Gets the types of all exceptions that can be thrown by a given method.
+     */
+    private static List<String> getExceptions(MethodUsage methodUsage) {
+        return methodUsage.getDeclaration().getSpecifiedExceptions()
+                .stream()
+                .map(JavaParserUtils::getTypeWithoutPackages)
+                .toList();
+    }
+
+    /**
+     * Gets the method signature from a JavaParser MethodUsage. Unfortunately,
+     * depending on the implementation of the ResolvedMethodDeclaration, it is
+     * not possible to recover specific features, such as:
+     * <ul>
+     *     <li>Modifiers</li>
+     *     <li>Annotations</li>
+     *     <li>Parameter names</li>
+     * </ul>
+     * This method considers three implementations of
+     * ResolvedMethodDeclaration: JavaParserMethodDeclaration,
+     * ReflectionMethodDeclaration, and JavassistMethodDeclaration. A
+     * signature follows the format:
+     *     "[modifiers] [typeParameters] [type] [methodName]([parameters]) throws [exceptions]"
+     */
+    public static String getMethodSignature(
+            MethodUsage methodUsage
+    ) {
+        ResolvedMethodDeclaration methodDeclaration = methodUsage.getDeclaration();
+        // Consider JavaParserMethodDeclaration.
+        if (methodDeclaration instanceof JavaParserMethodDeclaration jpMethodDeclaration) {
+            MethodDeclaration jpMethod = jpMethodDeclaration.getWrappedNode();
+            return getMethodSignature(jpMethod);
+        }
+        // Consider ReflectionMethodDeclaration or JavassistMethodDeclaration
+        Matcher matcher = METHOD_SIGNATURE.matcher(methodDeclaration.toString());
+        if (!matcher.find()) {
+            throw new IllegalStateException("Could not parse method signature: " + methodDeclaration);
+        }
+        String methodModifiers = methodDeclaration instanceof ReflectionMethodDeclaration ? matcher.group(1) : matcher.group(2);
```

Comment:
Put the `instanceof` test earlier, to choose the regular expression to use.  (Split METHOD_SIGNATURE into two separate regular expressions.)